### ISSUE TITLE
Bug: oneOf JSON parsing doesn't work

### ISSUE THREAD

**Issue by jamesshannon** [bufbuild/protobuf-es#434]
When you parse JSON your constructor insists on your non-standard `oneOf` object setup. You should instead accept something that's more inline with typical JSON outputs from other libraries. 

This is made more confusing because your `toJson()` function outputs the "typical" schema, which your `constructor` function rejects. Ie, now only can you not do _proto (from another package) -> JSON -> protobuf-es_, but you can't even do _protobuf-es -> JSON -> protobuf-es_. Here's a simple illustration:

```
const ft = new FieldType();
ft.fieldLocation.case = 'headerColumn';
ft.fieldLocation.value = 'why no work?';

const js = ft.toJson();
console.log(ft, js);

// This won't work. See the second row in the screenshot with `fieldLocation: {case: undefined}`
console.log(new FieldType(js));

// I need to manually modify protobuf-es' output to convert the oneof to something protobuf-es wants
js['fieldLocation'] = {case: 'headerColumn', value: 'now u works?'};

console.log(js, new FieldType(js));
```

Output:

![image](https://user-images.githubusercontent.com/761381/225500445-d85e47cd-cf3e-4ebc-8ab3-41216cbca1d4.png)

**Comment by jamesshannon** [comment]
I understand the reason, but it's, IMO, more trouble than it's worth. I'm currently trying to wire up a oneOf to different fields in a web-browser, and the need to do a translation layer to/from the current approach instead of simply _binding_ the fields. 

In any case, I read the docs, but I assumed that the "initializer object" in [the constructor](https://github.com/bufbuild/protobuf-es/blob/main/docs/runtime_api.md#constructing-messages) was also JSON and effectively the same as the "JSON Value" in the `fromJson()` class method that you pointed out. It seems I missed that. Also, for the record, it also handles the conversion of from sub-messages as well, which is good.

**Comment by timostamm** [comment]
@jamesshannon, `oneof` means that only one of the fields in a group can be present at a time. 

If you bind from some input field in the browser, you can't just set this field - you also have to unset all other fields in the group. The representation we use is an algebraic data type that guarantees correctness on the type level.

**Comment by jamesshannon** [comment]
Yes, i understand what `oneof` means.  However, I maintain that the way you've implemented it is harder to work with than what other languages do. Maybe that's partly a limitation of JS and the vision to "guarantee correctness on the type level", but it is a pain...

You must be aware of how this is handled by the Google protobuf generator for [go](https://protobuf.dev/reference/go/go-generated/#oneof), [java](https://protobuf.dev/reference/java/java-generated/#oneof), [python](https://protobuf.dev/reference/python/python-generated/#oneof), etc. None of them use the pattern that protobuf-es uses. In python's case, for example, it clears out all but the last field. Even if protobuf-es generated the `x` fields and then also a `.oneof_name_case` field which held the enum then that'd be easier then what it does today.
